src/core/src/core_logic/PatchAssessor.py

```diff
@@ -94,94 +103,109 @@ def raise_if_agent_incompatible(self):
-            return
+    def write_assessment_state(self):
+        self.composite_logger.log_debug("Updating assessment state... ")
 
-        self.composite_logger.log_debug("Updating core sequence... [Completed={0}]".format(str(completed)))
-        core_sequence = {'number': self.execution_config.sequence_number,
-                         'action': self.execution_config.operation,
-                         'completed': str(completed),
+        # lastHeartbeat below is redundant, but is present for ease of debuggability
+        assessment_state = {'number': self.execution_config.sequence_number,
+                         'lastStartInSecondsSinceEpoch': self.get_seconds_since_epoch(),
                          'lastHeartbeat': str(self.env_layer.datetime.timestamp()),
-                         'processIds': [os.getpid()] if not completed else [],
+                         'processIds': [os.getpid()],
                          'autoAssessment': str(self.execution_config.exec_auto_assess_only)}
-        core_state_payload = json.dumps({"coreSequence": core_sequence})
+        assessment_state_payload = json.dumps({"assessmentState": assessment_state})
 
         if os.path.isdir(self.assessment_state_file_path):
-            self.composite_logger.log_error("Core state file path returned a directory. Attempting to reset.")
+            self.composite_logger.log_error("assessment state file path returned a directory. Attempting to reset.")
             shutil.rmtree(self.assessment_state_file_path)
 
         for i in range(0, Constants.MAX_FILE_OPERATION_RETRY_COUNT):
             try:
                 with self.env_layer.file_system.open(self.assessment_state_file_path, 'w+') as file_handle:
-                    file_handle.write(core_state_payload)
+                    file_handle.write(assessment_state_payload)
             except Exception as error:
                 if i < Constants.MAX_FILE_OPERATION_RETRY_COUNT - 1:
-                    self.composite_logger.log_warning("Exception on core sequence update. [Exception={0}] [RetryCount={1}]".format(repr(error), str(i)))
+                    self.composite_logger.log_warning("Exception on assessment state update. [Exception={0}] [RetryCount={1}]".format(repr(error), str(i)))
                     time.sleep(i + 1)
                 else:
-                    self.composite_logger.log_error("Unable to write to core state file (retries exhausted). [Exception={0}]".format(repr(error)))
+                    self.composite_logger.log_error("Unable to write to assessment state file (retries exhausted). [Exception={0}]".format(repr(error)))
                     raise
 
-        self.composite_logger.log_debug("Completed updating core sequence.")
-
+        self.composite_logger.log_debug("Completed updating assessment state.")
+
+    def get_seconds_since_epoch(self):
+        return int((datetime.datetime.now() - datetime.datetime(1970, 1, 1)).total_seconds())
```

Comment:
I don't think that works like you think it works, my guess would be that datetime(1970, 1, 1) is in your local TZ? For example for me that resolves to -3600 epoch :)
But what you are trying to do is redundant anyways, epoch is literally the time in seconds since 1.1.1970, so you can just convert now into a timestamp. datetime.datetime.now().timestamp()